### PR TITLE
EPs Code review for Calculator

### PR DESCRIPTION
**Pull request by AnEvilPenguin**
A fleshed out version of the calculator project with all challenges attempted.

Thanks for your time!

### CODE REVIEW

File: AnEvilPenguin.Calculator/CalculatorLibrary/CalculatorLibrary.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+            writer.WriteEndArray();
+            writer.WriteEndObject();
+
+            pastResults.Clear();
+        }
+
+        public double DoOperation(double num1, double num2, string op)
+        {
+            double result = double.NaN; // Default value is "not-a-number" if an operation, such as division, could result in an error.
```
Comment (TheCSharpAcademy) [inline AnEvilPenguin.Calculator/CalculatorLibrary/CalculatorLibrary.cs:38]: It's totally fine to make comments when you're studying, but whenever pushing code make sure your class members ( methods, fields, properties) and properties are named in a way that makes your code self-explanatory. Only make comments when absolutely necessary and there's something that can't be inferred from code.  Please review all comments and only leave the ones that are absolutely necessary for other devs (or your future self) :-)

File: AnEvilPenguin.Calculator/CalculatorLibrary/CalculatorLibrary.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
+﻿using Newtonsoft.Json;
+using System.Diagnostics;
```
Comment (TheCSharpAcademy) [inline AnEvilPenguin.Calculator/CalculatorLibrary/CalculatorLibrary.cs:2]: Please remove unused using. 

image

**Review by TheCSharpAcademy — CHANGES_REQUESTED** [review]

File: AnEvilPenguin.Calculator/CalculatorLibrary/CalculatorLibrary.cs (hunk 1 above)
Comment (AnEvilPenguin) [reply AnEvilPenguin.Calculator/CalculatorLibrary/CalculatorLibrary.cs]: Makes sense!

File: AnEvilPenguin.Calculator/CalculatorLibrary/CalculatorLibrary.cs (hunk 2 above)
Comment (AnEvilPenguin) [reply AnEvilPenguin.Calculator/CalculatorLibrary/CalculatorLibrary.cs]: Done! Annoyed I missed that!

**Comment by AnEvilPenguin** [comment]
Changes made. Thanks!

**Pull request merged**